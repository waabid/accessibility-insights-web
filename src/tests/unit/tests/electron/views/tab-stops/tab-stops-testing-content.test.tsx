// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { TabStopsActionCreator } from 'electron/flux/action/tab-stops-action-creator';
import {
    TabStopsTestingContent,
    TabStopsTestingContentProps,
} from 'electron/views/tab-stops/tab-stops-testing-content';
import { shallow } from 'enzyme';
import { Toggle } from 'office-ui-fabric-react';
import * as React from 'react';
import { IMock, Mock, MockBehavior } from 'typemoq';

describe('TabStopsTestingContent', () => {
    let tabStopsActionCreatorMock: IMock<TabStopsActionCreator>;
    let props: TabStopsTestingContentProps;

    beforeEach(() => {
        tabStopsActionCreatorMock = Mock.ofType<TabStopsActionCreator>(
            undefined,
            MockBehavior.Strict,
        );
        props = {
            deps: { tabStopsActionCreator: tabStopsActionCreatorMock.object },
            showTabStops: true,
        };
    });

    test('renders', () => {
        const testSubject = shallow(<TabStopsTestingContent {...props} />);
        expect(testSubject.getElement()).toMatchSnapshot();
    });

    test('toggles tab stops off', () => {
        tabStopsActionCreatorMock.setup(m => m.disableTabStops()).verifiable();
        const testSubject = shallow(<TabStopsTestingContent {...props} />);
        const onToggle = testSubject.find(Toggle).prop('onClick');
        const eventStub = null;
        onToggle(eventStub);

        tabStopsActionCreatorMock.verifyAll();
    });

    test('toggles tab stops on', () => {
        props.showTabStops = false;
        tabStopsActionCreatorMock.setup(m => m.enableTabStops()).verifiable();
        const testSubject = shallow(<TabStopsTestingContent {...props} />);
        const onToggle = testSubject.find(Toggle).prop('onClick');
        const eventStub = null;
        onToggle(eventStub);

        tabStopsActionCreatorMock.verifyAll();
    });
});
