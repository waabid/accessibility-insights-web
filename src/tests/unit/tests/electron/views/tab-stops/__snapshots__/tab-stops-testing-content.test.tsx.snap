// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabStopsTestingContent renders 1`] = `
<React.Fragment>
  <StyledToggleBase
    checked={true}
    className="toggle"
    label="Show tab stops"
    offText="Off"
    onClick={[Function]}
    onText="On"
  />
  <span
    aria-level={3}
    role="heading"
  >
    <strong>
      How to test:
    </strong>
  </span>
  <ol
    className="howToTestList"
  >
    <li>
      Navigate to the screen you want to test in your app.
    </li>
    <li>
      Turn on the Show tab stops toggle. An empty circle will highlight the element with focus.
    </li>
    <li>
      If you are connected to an emulator, select your emulator app to bring it into focus and use your keyboard.
    </li>
    <li>
      Or If you are connected to a physical mobile device use the keyboard key-buttons below.
    </li>
    <li>
      Move input focus through all the interactive elements in the page:
      <ul>
        <li>
          Use Tab and Shift+Tab to navigate between standalone controls.
        </li>
        <li>
          Use the arrow keys to navigate between the focusable elements within a composite control.
        </li>
      </ul>
    </li>
    <li>
      As you navigate to each element, look for these accessibility problems:
      <ul>
        <li>
          An interactive element can't be reached using the Tab and arrow keys.
        </li>
        <li>
          An interactive element "traps" input focus and prevents navigating away.
        </li>
        <li>
          An interactive element doesn't give a visible indication when it has input focus.
        </li>
        <li>
          The tab order is inconsistent with the logical order that's communicated visually.
        </li>
        <li>
          Input focus moves unexpectedly without the user initiating it.
        </li>
      </ul>
    </li>
  </ol>
</React.Fragment>
`;
